adagrams/game.py

```diff
@@ -1,11 +1,73 @@
+from random import shuffle
+from adagrams import LETTER_POOL, SCORE_CHART
+from copy import deepcopy
+
+def create_letter_pool():
+    letter_pool = []
+    for letter, frequency in LETTER_POOL.items():
+        for i in range(frequency):
+            letter_pool.append(letter)
+    
+    return letter_pool
+
+
 def draw_letters():
-    pass
+
+    letter_pool = create_letter_pool()
+    # shuffle method from random module allows us to randomize the list in-place
+    shuffle(letter_pool)
+    # once the list is shuffled, we slice the first 10 items to use as the user's letter bank
+    letter_bank = letter_pool[0:10]
+
+    return letter_bank
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    
+    # make a copy of letter bank list to avoid editing original list
+    copy_of_letter_bank = deepcopy(letter_bank)
+
+    # check if each letter in upper_case word is in letter_bank
+    # remove the letter if identified to update occurrence frequency in list, letter_bank
+    for letter in word.upper():
+        if letter not in copy_of_letter_bank:
+            return False
+        else:
+            copy_of_letter_bank.remove(letter)
```

Comment:
Thanks for this feedback, Ansel! Does the time complexity change since we know that the length of `word` will never exceed 10? I guess what I'm asking is, can we reduce the potential time complexity by including guardrails in our code that limits the length of what we're iterating over?

And in terms of alternative data structures, would a dictionary be appropriate here? The idea being that we could access the exact key of the dictionary (i.e., the letter) and reduce its value by 1?